enclave/core/sgx/tracee.c

```diff
@@ -3,19 +3,58 @@
 
 #include <openenclave/bits/sgx/sgxtypes.h>
 #include <openenclave/enclave.h>
+#include <openenclave/internal/raise.h>
 #include <openenclave/internal/report.h>
 #include <openenclave/internal/sgx/td.h>
 
 #include "../tracee.h"
 #include "report.h"
+#include "tracee.h"
 
 static volatile int _is_enclave_debug_allowed = -1;
+static volatile int _is_in_simulation_mode = -1;
+
+oe_result_t oe_sgx_initialize_simulation_mode_cache(oe_sgx_td_t* td)
```

Comment:
Do we need this mechanism for td->simulate? Is that field ever cleared by the enclave?

Reply:
FS[0] is initialized as part of `td_init` (https://github.com/openenclave/openenclave/blob/401ef13602a560f5719eb951ef6056f3fc0b97d2/enclave/core/sgx/td_basic.c#L96), which is invoked in the first invocation of `_handle_ecall`. However, the invocation of `oe_is_enclave_debug_allowed` used by the ecall stitching is earlier that that (https://github.com/openenclave/openenclave/blob/401ef13602a560f5719eb951ef6056f3fc0b97d2/enclave/core/sgx/calls.c#L1198)

Reply:
Got it. I was thinking of oe_enter where we are able to access td fields like host_rbp etc, even before td is initialized. But that has td.

Should we setup FS[0], early enough in oe_enter? The value of FS is guaranteed to be OE's vFS value.